memuvarint.go

```diff
@@ -44,6 +44,11 @@ func (r *memUvarintReader) ReadUvarint() (uint64, error) {
 	var C = r.C
 	var S = r.S
 
+	if C >= len(S) {
```

Comment:
Minor thing - this check could be the first thing in this function so that lines 42-45 become redundant in such cases.

Reply:
Ack.